addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -677,16 +677,28 @@ var FileWidget = SearchableMediaWidget.extend({
     /**
      * @private
      */
-    _onURLInputChange: function () {
+    _onURLInputChange: async function() {
         const inputValue = this.$urlInput.val().split('?')[0];
         var emptyValue = (inputValue === '');
 
-        var isURL = /^.+\..+$/.test(inputValue); // TODO improve
-        var isImage = _.any(this.IMAGE_EXTENSIONS, function (format) {
-            return inputValue.endsWith(format);
-        });
-
-        this._updateAddUrlUi(emptyValue, isURL, isImage);
+        const isURL = /^(?:(?:https?|http):\/\/)?(?:\S+(?::\S*)?@)?(?:www\.)?(?:[-\w.]+)(?::\d+)?(?:\/[\S]*)?$/.test(inputValue);
+        var isImage = false;
+        if (isURL) {
+            await fetch(inputValue)
```

Comment:
But this is going to fetch the _entire_ file on input change, so if it's a big file it's quickly going to become an issue. I reread the discussion in the task chatter and now I'm a little confused. @mart-e This can't be what you meant, right ? ^^'